packages/functions/src/startTest.ts

```diff
@@ -18,64 +19,97 @@ const dynamoDb = DynamoDBDocumentClient.from(client);
 export const main = async (
   event: APIGatewayProxyEvent,
 ): Promise<APIGatewayProxyResult> => {
-  const PK = event.pathParameters?.questionType;
-  const possibleQuestionTypes = [
-    'writing',
-    'reading',
-    'listening',
-    'speaking',
-    'fullTest',
-  ];
+  //   const PK = event.pathParameters?.questionType;
+  //   const possibleTestTypes = [
+  //     'writing',
+  //     'reading',
+  //     'listening',
+  //     'speaking',
+  //     'fullTest',
+  //   ];
 
   // Test sections
   const testSections = ['writing', 'reading', 'listening', 'speaking'];
 
-  const userID = event.requestContext.authorizer;
-  console.log(userID);
-
-  //validate the question type
-  if (!PK || !possibleQuestionTypes.includes(PK)) {
+  const userID = event.requestContext.authorizer!.jwt.claims.sub;
+  if (!userID) {
     return {
       statusCode: 400,
       body: JSON.stringify({
-        message: 'Invalid question type',
+        message: 'Invalid user ID',
       }),
     };
   }
 
+  //validate the question type
+  //   if (!PK || !possibleTestTypes.includes(PK)) {
+  //     return {
+  //       statusCode: 400,
+  //       body: JSON.stringify({
+  //         message: 'Invalid question type',
+  //       }),
+  //     };
+  //   }
+
   try {
     // Get the index of the questions
-    let Questions;
-    if (PK === 'FullTest') {
-      const _Questions = testSections.map(async (PK: string) => {
-        return await getQuestion(PK);
-      });
-      const QuestionsArray = Promise.all(_Questions);
-      Questions = {
-        ...(await QuestionsArray)[0],
-        ...(await QuestionsArray)[1],
-        ...(await QuestionsArray)[2],
-        ...(await QuestionsArray)[3],
-      };
-    } else {
-      Questions = await getQuestion(PK);
-    }
+    let questions;
+    // if (PK === 'fullTest') {
+    const _Questions = testSections.map(async (PK: string) => {
+      return await getQuestion(PK);
+    });
+    const QuestionsArray = Promise.all(_Questions);
+    questions = {
+      writing: (await QuestionsArray)[0],
+      reading: (await QuestionsArray)[1],
+      listening: (await QuestionsArray)[2],
+      speaking: (await QuestionsArray)[3],
+    };
```

Comment:
Why not use await once in the first line?
```ts
const QuestionsArray = await Promise.all(_Questions);
```

Reply:
Noted